src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -451,3 +456,52 @@ export const handleFormAuthRedirect = [
   }),
   _handleFormAuthRedirect,
 ] as RequestHandler[]
+
+/**
+ * Handler for validating the eServiceId of a given form
+ *
+ * @returns 200 with eserviceId validation result
+ * @returns 400 when there is an error on the authType of the form
+ * @returns 400 when the eServiceId of the form does not exist
+ * @returns 404 when form with given ID does not exist
+ * @returns 500 when the title of the fetched login page does not exist
+ * @returns 500 when database error occurs
+ * @returns 500 when the url for the login page of the form could not be generated
+ * @returns 502 when the login page for singpass could not be fetched
+ */
+export const handleValidateFormEsrvcId: RequestHandler<
+  { formId: string },
+  PublicFormAuthValidateEsrvcIdDto | ErrorDto
+> = (req, res) => {
+  const { formId } = req.params
+  return FormService.retrieveFormById(formId)
+    .andThen((form) => {
+      // NOTE: Because the check is based on parsing the html of the returned webpage,
+      // And because MyInfo login is beyond our control, we coerce MyInfo to SP.
+      // This is valid because a valid MyInfo eserviceId is also a valid SP eserviceId
+      if (form.authType === AuthType.MyInfo) {
+        return validateMyInfoForm(form).andThen((form) =>
+          SpcpFactory.createRedirectUrl(AuthType.SP, formId, form.esrvcId),
+        )
+      }
+      return validateSpcpForm(form).andThen((form) =>
+        SpcpFactory.createRedirectUrl(form.authType, formId, form.esrvcId),
+      )
```

Comment:
(i know this is wip, just happened to see it)

should we split up the steps to 
1. a validate step and
2. the creation of redirect url step so we don't need to repeat the same steps twice?

```ts
// redirect step
const redirectAuthType = form.authType === AuthType.MyInfo ? AuthType.SP : form.authType
return SpcpFactory.createRedirectUrl(redirectAuthType, formId, form.esrvcId),
```

Reply:
didn't do this because i wanted the coercion from `MyInfo` to `SP` to be clear. although it's slightly longer, i think the explicitness is worth the length increase!